From c429024362f1b08275595dbb1c74a211e149f2d6 Mon Sep 17 00:00:00 2001
From: Jactry Zeng <jzeng@codeweavers.com>
Date: Sun, 26 Jun 2016 12:03:20 +0800
Subject: [PATCH] dwrite: Hack: Add font replacement for Meiryo.
To: wine-patches <wine-patches@winehq.org>
Reply-To: wine-devel <wine-devel@winehq.org>

This is being done as a hack in the code rather than by changing the registry setting
for several reasons:

1) Registry setting is platform-specific, but bottles can be moved between Mac, Linux.

2) We are using 'Meiryo' for all CJK now, but in future we may use different fonts
for different system language. Then we will need different replacements for them.

Signed-off-by: Jactry Zeng <jzeng@codeweavers.com>
---                                                                                                     
 dlls/dwrite/font.c | 5 +++++                                                                           
 1 file changed, 5 insertions(+)                                                                        
                                                                                                        
diff --git a/dlls/dwrite/font.c b/dlls/dwrite/font.c                                                    
index b625f7a..aa92548 100644                                                                           
--- a/dlls/dwrite/font.c                                                                                
+++ b/dlls/dwrite/font.c
@@ -3437,6 +3437,9 @@ static void fontcollection_add_replacements(struct dwrite_fontcollection *collec
     WCHAR *name;
     void *data;
     HKEY hkey;
+    WCHAR meiryoW[] = {'M','e','i','r','y','o',0};
+    WCHAR meiryo_replacement[] = {'I','n','z','i','u',' ','R','o','b','o','t','o',' ','S','C'};
+
 
     if (RegOpenKeyA(HKEY_CURRENT_USER, "Software\\Wine\\Fonts\\Replacements", &hkey))
         return;
@@ -3475,6 +3478,8 @@ static void fontcollection_add_replacements(struct dwrite_fontcollection *collec
     heap_free(data);
     heap_free(name);
     RegCloseKey(hkey);
+
+    fontcollection_add_replacement(collection, meiryoW, meiryo_replacement);
 }
 
 HRESULT create_font_collection(IDWriteFactory3 *factory, IDWriteFontFileEnumerator *enumerator, BOOL is_system, IDWriteFontCollection **ret)
-- 
2.4.9 (Apple Git-60)
